code/modules/instruments/items.dm

```diff
@@ -1,63 +1,58 @@
 //copy pasta of the space piano, don't hurt me -Pete
 /obj/item/instrument
 	name = "generic instrument"
-	resistance_flags = FLAMMABLE
 	force = 10
 	max_integrity = 100
+	resistance_flags = FLAMMABLE
 	icon = 'icons/obj/musician.dmi'
 	lefthand_file = 'icons/mob/inhands/equipment/instruments_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/equipment/instruments_righthand.dmi'
+	/// Our song datum.
 	var/datum/song/handheld/song
-	var/instrumentId = "generic"
-	var/instrumentExt = "mid"
-	var/instrumentRange = 15
+	/// Our allowed list of instrument ids. This is nulled on initialize.
+	var/list/allowed_instrument_ids
+	/// How far away our song datum can be heard.
+	var/instrument_range = 15
 
-/obj/item/instrument/Initialize()
+/obj/item/instrument/Initialize(mapload)
 	. = ..()
-	song = new(instrumentId, src, instrumentExt, instrumentRange)
+	song = new(src, allowed_instrument_ids, instrument_range)
+	allowed_instrument_ids = null			//We don't need this clogging memory after it's used.
 
 /obj/item/instrument/Destroy()
 	QDEL_NULL(song)
-	. = ..()
+	return ..()
+
+/obj/item/instrument/proc/should_stop_playing(mob/user)
+	return !user.CanReach(src) || !user.canUseTopic(src, FALSE, TRUE, FALSE, FALSE)
```

Comment:
Here is the issue I think. canUseTopic()

In this file https://github.com/tgstation/tgstation/blob/master/code/modules/mob/living/living.dm#L994
canUseTopic returns false if the user is incapacitated, which I assume is the case when you're prone for whatever reason. I dunno how you'd solve this other than changing the proc, or doing a snowflake alternative, but it has to be solved.